cmd/mdatagen/templates/status.go.tmpl

```diff
@@ -6,4 +6,7 @@ import (
     "go.opentelemetry.io/collector/component"
 )
 
-const Stability = component.StabilityLevel{{ casesTitle .Status.Stability }}
\ No newline at end of file
+const (
+	Name = "{{ .Name }}"
```

Comment:
I was thinking about this change earlier today, I wonder if there's a way to use the name as a replacement for `typeStr` field which is defined in most components. I think we may be able to use `name` in 3 different places:

1.  to remove the need for the typeStr field
2. to replace the header in the readme
3. combining name + type in the metadata to produce the identifier many components (mostly scrapers) are using today in their scope names.

I would imagine a receiver like the `Oracle DB` contain the following config:

```yaml
name: Oracle DB
status:
  type: receiver
```

- In the header of the readme, the usage could be: `{{ .Name }} {{ casesTitle .Status.Type }}`
- In `typeStr`, the name could be lowered + any spaces stripped.
- In the name used in the Scope, it would be a combination of the `typeStr` + the `type`. 

I only looked at about 10 or so components when I was wondering if this could work, and i think it does. What do you think?

Reply:
I thought about this. I actually started coding towards it and stopped so I could ship this more narrow set to address 1.

If you want 2, maybe we can do it with a “label” entry.

For 3, definitely. I wasn’t sure if we should go there but it makes sense.

Reply:
My only concern w/ the pr as is, is that it adds another identifier with a slightly different name from the `typeStr`. If we could combine the name + type, then `name` could effectively be `typeStr`, and the combination of `typeStr` + the `type` metadata field could form the scope name.

Reply:
Sorry for being dense, I understand what you're asking for, I think. I'll add that.

Reply:
> I was thinking about this change earlier today, I wonder if there's a way to use the name as a replacement for `typeStr` field which is defined in most components. I think we may be able to use `name` in 3 different places:
> 
> 1. to remove the need for the typeStr field
> 2. to replace the header in the readme
> 3. combining name + type in the metadata to produce the identifier many components (mostly scrapers) are using today in their scope names.

I think there is a problem with using "Type" and "Name" like this. 

[`component.ID` consists of `type` and `name`](https://github.com/open-telemetry/opentelemetry-collector/blob/ff0dd4ce848dc6583609452f114f4b6eca423fb5/component/identifiable.go#L41), with `name` being optional. This is or should be user facing terminology in my opinion. I'd rather we preserve these semantics and find other things to change.


> I would imagine a receiver like the `Oracle DB` contain the following config:
> 
> ```yaml
> name: Oracle DB
> status:
>   type: receiver
> ```

What do you think of introducing the term `class` to describe the broad categories of `receiver`, `processor`, `exporter`, `connector`?  This would give us:

```yaml
type: Oracle DB
status:
  class: receiver
```

I believe we can still autogenerate all the same things, but we'll have a lot less overloading of terminology. WDYT?

Reply:
This works for me. Thanks for catching this @djaglowski

Reply:
I'm not totally following, isn't Type already the keyword for denoting Receiver/Processor/Exporter/Extension?

Reply:
Ok I'm following now, thanks!

Reply:
ok so changing to use:
```
type: Oracle DB
status:
  class: receiver
```